framework/src/systems/SystemBase.C

```diff
@@ -823,6 +823,9 @@ SystemBase::addVariable(const std::string & var_type,
     // The number returned by libMesh is the _last_ variable number... we want to hold onto the
     // _first_
     var_num = system().add_variables(var_names, fe_type, &blocks) - (components - 1);
+
+    // Set as array variable
+    parameters.set<bool>("array_variable") = true;
```

Comment:
```suggestion
    parameters.set<bool>("array") = true;
```

Reply:
We should add a check right before here as well for if the user _explicitly_ set `false` as `array` but has `components` > 1:

```
if (parameters.isParamSetByUser("array") && !parameters.get<bool>("array"))
  mooseError("Variable '", name, "' is an array variable ('components' > 1) but 'array' is set to false.");
```

Of course, if they _didn't_ set it, it's okay because we're gonna be setting it to true.

You should be able to test for this with your current test spec with:
```
  [array_variable_array_false]
    type = 'RunException'
    input = 'array_variable_size_one_test.i'
    requirement = 'MOOSE shall report a reasonable error when defining a variable with multiple components not as an array variable.'
    issues = '#19564'
    expect_out = "'array' is set to false"
    design = 'ArrayMooseVariable.md'
  [../]
```

or something like that